packages/terra-tag/src/TagList.jsx

```diff
@@ -0,0 +1,342 @@
+   * attribute to the Tag List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If a visible text label is used with the collection of tags, provide a string of the ID for the html
+   * element containing the visible text label. Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel'
+   * prop and adds the `aria-labelledby` attribute instead to the Tag List container element.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) For best practices, ensure the visible text
+   * in the html element `id` provided to the Tag List `ariaLabelledby` prop matches the text provided to the
+   * `ariaLabel` prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that
+   * help describe the intent of the group of tags.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The content to be shown in the tags container.
+   */
+  children: PropTypes.node,
+  /**
+   * Indicates if the Tags container is rolled up or not.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  isCollapsible: false,
+};
+
+const TagList = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    intl,
+    isCollapsible,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const [containerTabindex, setContainerTabindex] = useState('-1');
+  const [updatedCount, setUpdatedCount] = useState(React.Children.count(children));
+  const [rollUpCount, setRollUpCount] = useState(React.Children.count(children));
+  const [isCollapsed, setIsCollapsed] = useState(isCollapsible);
+  const currentTag = useRef(); // ID of the tag that will receive focus
+  const filterTagsRef = useRef();
+  const focusNode = useRef(0);
+  const isRollUpRemoved = useRef(false);
+  const containerHint = useRef(ariaLabel);
+
+  // Identifies the number of tags that needs to be hidden/rolled up
+  const generateRollUp = useCallback(() => {
+    const startIndex = TagsUtils.getRollUpIndex(filterTagsRef);
+    if (isCollapsed) {
+      setUpdatedCount(startIndex);
+      setRollUpCount(React.Children.count(children) - startIndex);
+    } else if (startIndex === React.Children.count(children)) {
+      setRollUpCount(0);
+    }
+  }, [children, isCollapsed]);
+
+  // Modifies the tabindex of the tag
+  const setTabIndex = (val) => {
+    const currentNode = currentTag.current ? filterTagsRef.current.querySelector(`#${currentTag.current}`) : null;
+    if (currentNode) {
+      currentNode.setAttribute('tabIndex', val);
+    }
+  };
```

Comment:
We should create an appropriate state variable and update the tabIndex when the value changes.  We should not need to query the DOM since we are rendering the components, correct?

Reply:
Since we invoke this setTabIndex method on every keyboard navigation to toggle the focused tag tabindex using a state will re-render for every navigation we have used the ref here to updated the tabindex of focused tab. Thanks